analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,189 @@
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    var catchClause = (CatchClauseSyntax)c.Node;
+                    if (AzureFunctionEntryPoint(c.SemanticModel, catchClause, c.CancellationToken) is { } entryPoint)
+                    {
+                        var iLogger = c.SemanticModel.Compilation.GetTypeByMetadataName(KnownType.Microsoft_Extensions_Logging_ILogger.TypeName);
+                        if (iLogger is not null
+                            && entryPoint.Parameters.Any(p => p.Type.DerivesOrImplements(iLogger))
+                            && InvalidCallsToILogger(c.SemanticModel, iLogger, catchClause, c.CancellationToken) is ImmutableArray<ExpressionSyntax> secondaryLocations)
+                        {
+                            c.ReportIssue(Diagnostic.Create(Rule, catchClause.CatchKeyword.GetLocation(), additionalLocations: secondaryLocations.Select(x => x.GetLocation())));
+                        }
+                    }
+                },
+                SyntaxKind.CatchClause);
+
+        private static ImmutableArray<ExpressionSyntax>? InvalidCallsToILogger(SemanticModel semanticModel, ITypeSymbol iLogger, CatchClauseSyntax catchClause, CancellationToken cancellationToken)
+        {
+            var walker = new LoggerCallWalker(LanguageFacade, semanticModel, iLogger, cancellationToken);
+            if (catchClause.Block is { } block)
+            {
+                walker.Visit(block);
+            }
+
+            if (!walker.HasValidLoggerCall && catchClause.Filter?.FilterExpression is { } filterExpression)
+            {
+                walker.Visit(filterExpression);
+            }
+
+            return walker.HasValidLoggerCall
+                ? null
+                : walker.InvalidLoggerInvocations;
+        }
+
+        private static IMethodSymbol AzureFunctionEntryPoint(SemanticModel model, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            return node.Ancestors().OfType<MethodDeclarationSyntax>().FirstOrDefault() is { } method
+                && method.AttributeLists.Any() // FunctionName has [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)] so there must be an attribute
+                && model.GetDeclaredSymbol(method, cancellationToken) is { } methodSymbol
+                && model.Compilation.GetTypeByMetadataName(KnownType.Microsoft_Azure_WebJobs_FunctionNameAttribute.TypeName) is { } functionNameAttribute
+                && methodSymbol.GetAttributes().Any(a => a.AttributeClass.Equals(functionNameAttribute))
+                ? methodSymbol
+                : null;
+        }
+
+        private sealed class LoggerCallWalker : SafeCSharpSyntaxWalker
+        {
+            private static readonly int[] ValidLogLevel = new[]
```

Comment:
The "LogInformation", .. method names should go also on top then, shouldn't they?

Reply:
They could, if they would be extracted in a field. The current switch stuff is practical as it is, because there's quite some complicated logic around it.